src/control/cmd/agent/infocache.go

```diff
@@ -58,22 +61,27 @@ type attachInfoCache struct {
 // to choose from.  Returns the index of the device to use.
 func (aic *attachInfoCache) loadBalance(numaNode int) int {
```

Comment:
Just a thought while reviewing this, not something that needs to or should be changed now... I wonder if it would be adequate to just randomly pick a device so that you could get rid of the write locking. Maybe something to explore in the future if this turns out to be a hot spot.